crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -658,14 +703,43 @@ where
                 // channel according to its
                 // `max_inbound_htlc_value_in_flight_percent_of_channel`
                 // configuration value
+
+                let user_channel_id = Uuid::new_v4().as_u128();
                 let temp_channel_id = match self.channel_manager.create_channel(
                     target_node_id,
                     channel_value,
                     0,
-                    0,
+                    user_channel_id,
                     Some(user_config),
                 ) {
-                    Ok(temp_channel_id) => temp_channel_id,
+                    Ok(temp_channel_id) => {
+                        let user_channel_id = Uuid::from_u128(user_channel_id).to_string();
+                        tracing::debug!(
+                            "Creating shadow channel with user_channel_id: {user_channel_id}"
+                        );
+                        // We create a shadow copy of the channel as the the ldk channel does not
+                        // live beyond channel closure. The main purpose of
+                        // this shadow is to track general meta data of the
+                        // channel relevant for reporting purposes.
+                        let new_channel =
+                            Channel::new(user_channel_id, channel_value as i64, target_node_id);
+                        if let Err(err) = self.storage.upsert_channel(new_channel) {
+                            // The channel is already in creation so we need to close the channel.
+                            // The closed channel should then fail the intercepted payment. We could
+                            // already fail the payment here, but it's nicer to have the logic
+                            // consistent at one place, when processing the `Event:ChannelClosed`
+                            tracing::error!("Failed to insert channel to database. Closing channel with temp_channel_id {}! Error: {err:?}", hex::encode(temp_channel_id));
+                            // Note, we can use the temp_channel_id here, as the channel is still in
+                            // creation.
+                            if let Err(err) = self
```

Comment:
🔧 its probably better to fail the htlc intercepted instead of closing the channel immediately after creating it. 

In other words, I am going to change that so that the shadow channel will get created before we actually create the channel and fail the intercepted htlc if unsuccessful.